src/main/java/be/seeseemelk/mockbukkit/inventory/meta/EnchantedBookMetaMock.java

```diff
@@ -82,7 +85,13 @@ public boolean addStoredEnchant(@NotNull Enchantment ench, int level, boolean ig
 		}
 
 		Integer prev = storedEnchantments.put(ench, level);
-		return prev == null || prev.intValue() != level;
+		return prev == null || prev != level;
```

Comment:
Revert this please.
```suggestion
		return prev == null || prev.intValue() != level;
```

Reply:
Why ?
https://stackoverflow.com/a/11490442

Reply:
You should perhaps read the thread you posted.
Because it precisely tells you why this is wrong...

The `==` operator is **not** an equality check, it is a memory address comparison.
Only for primitives can it be used to check for equality but not for Objects. The `Integer` class is what we call a "boxed primitive", it's a wrapper for a primitive but itself is still an `Object` and will follow the object's rules.
Two objects will only ever be equal if they are stored under the same memory location.
Now there are some exceptions due to JVM optimizations which will for example put literals and constants under the same memory location sometimes. But for almost all production code the usage of `==` for comparing Objects should be avoided **at all cost**. It is a really bad practice. And that also explains for boxed integers.
The JVM may still do an implict type transformation to make this work sometimes but it is inherently flawed and should not be relied upon.
That's why this change needs to reverted immediately, we need to make sure that this is correctly converted into a primitive.